picasso-compose/src/main/java/com/squareup/picasso3/compose/PicassoPainter.kt

```diff
@@ -71,20 +78,23 @@ internal class PicassoPainter(
     }
   }
 
-  override fun onRemembered() {
-    request.invoke(picasso).into(this)
+  fun start() {
+    stateObserver.start()
+    loadRequest()
   }
 
-  override fun onAbandoned() {
-    (painter as? RememberObserver)?.onAbandoned()
+  fun stop() {
+    stateObserver.stop()
+    stateObserver.clear()
+    (painter as? RememberObserver)?.onForgotten()
     painter = EmptyPainter
     picasso.cancelRequest(this)
   }
 
-  override fun onForgotten() {
-    (painter as? RememberObserver)?.onForgotten()
-    painter = EmptyPainter
-    picasso.cancelRequest(this)
+  private fun loadRequest() {
+    stateObserver.observeReads(this, onCommitAffectingRequest) {
+      request.invoke(picasso).into(this)
```

Comment:
Technically you only need to call `request.invoke(picasso)` inside the `observeReads` lambda. However, since we know `into` isn't doing any state reads we don't care about, it doesn't really matter.

If `RequestCreator`s are comparable, you might want to wrap the `request.invoke(picasso)` call in a `derivedStateOf` stored in a class property to avoid making new requests for equivalent `RequestCreator`s.